llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -132,6 +132,11 @@ cl::opt<bool> AtomicFirstCounter(
              "the entry counter)"),
     cl::init(false));
 
+cl::opt<bool> ConditionalCounterUpdate(
+    "conditional-counter-update",
+    cl::desc("Do conditional counter updates in single byte counters mode)"),
+    cl::init(false));
```

Comment:
Is there any reason why this shouldn't be enabled by default?
```suggestion
    cl::init(true));
```

Reply:
`lowerCover()` was originally introduced as a binary size optimization. If we make this the default we will regress size quite a bit, so I would rather keep the default `false`.

Reply:
I did not enable it by default in case there are any cases where this might not beneficial, and @ellishg raised a good point why we should not make it by default because of the original motivation of single byte counters in PGO.